### PR TITLE
Implement handlers

### PR DESCRIPTION
**Pull request by dpordomingo**
partially solves #3
supersedes https://github.com/src-d/code-annotation/pull/16

edd1807 Implement `/experiment` handlers

### CODE REVIEW

**Comment by dpordomingo** [comment]
@carlosms you can take this as a dependency of your work with the repositories till @bzz approves it

File: server/handler/assignments.go [hunk 1]
```diff
@@ -1,17 +1,43 @@
...
 func GetAssignmentsForUserExperiment() RequestProcessFunc {
 	return func(r *http.Request) (*serializer.Response, error) {
-		return nil, serializer.NewHTTPError(http.StatusNotImplemented)
+		requestedExperimentID := chi.URLParam(r, "experimentId")
+		experimentID, err := strconv.Atoi(requestedExperimentID)
+		if err != nil {
+			return nil, fmt.Errorf("wrong format in experiment ID sent; received %s", requestedExperimentID)
+		}
+
+		// TODO: fetch from user in session
```
Comment (smacker) [inline server/handler/assignments.go]: why don't use `jwt.GetUserID`? Middleware is already there.

File: server/handler/assignments.go [hunk 2]
```diff
@@ -1,17 +1,43 @@
...
+		userID, err := strconv.Atoi("1")
+		if err != nil {
+			return nil, fmt.Errorf("wrong format in user ID sent; received %s", "1")
+		}
+
+		assignments, err := repository.GetAssignmentsFor(userID, experimentID)
+		if err == repository.ErrNoAssignmentsInitialized {
+			if assignments, err = repository.CreateAssignmentsFor(userID, experimentID); err != nil {
+				return nil, fmt.Errorf("no available assignments")
```
Comment (smacker) [inline server/handler/assignments.go]: the original error should be logged (ok to do in separate PR)

File: server/handler/assignments.go [hunk 3]
```diff
@@ -23,6 +49,27 @@ type assignmentRequest struct {
 // SaveAssignment returns a function that saves the user answers as passed in the body request
 func SaveAssignment() RequestProcessFunc {
 	return func(r *http.Request) (*serializer.Response, error) {
-		return nil, serializer.NewHTTPError(http.StatusNotImplemented)
+		requestedAssignmentID := chi.URLParam(r, "assignmentId")
+		assignmentID, err := strconv.Atoi(requestedAssignmentID)
+		if err != nil {
+			return nil, fmt.Errorf("wrong format in assignment ID sent; received %s", requestedAssignmentID)
```
Comment (smacker) [inline server/handler/assignments.go]: it's not 500 but 404 or 400. Ok to do in separate PR.

**Comment by smacker** [comment]
also CI is failing with error
```
server/handler/experiments.go:8:2: cannot find package "github.com/src-d/code-annotation.bak/server/repository"
```

File: server/handler/assignments.go (hunk 2 above)
Comment (dpordomingo) [reply server/handler/assignments.go]: I'll do so by https://github.com/src-d/code-annotation/issues/50

**Comment by dpordomingo** [comment]
@smacker your comments were addressed :dancer:

**Review by smacker — APPROVED** [review]

**Review by bzz — APPROVED** [review]
LGTM!
